### PR TITLE
remove external_variables

### PR DESCRIPTION
**Pull request by TomTranter**
# Description

Remove external_variables which have functionality that is now completely replicable with input parameters. Motivation is to simplify code and reduce redundant functionality. Also relates to a liionpack issue with handling the two types of data seperately and is cleaner to use inputs.

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: tests/integration/test_models/test_full_battery_models/test_lithium_ion/test_external/test_external_current_collector.py [hunk 1]
```diff
@@ -1,51 +1,54 @@
...
-                "Current collector current density": i_boundary_cc,
-            }
-
-            sim.step(dt, external_variables=external_variables)
-
-            # obtain phi_s_n from the pybamm solution at the current time
-            phi_s_p = sim.solution["Positive current collector potential"].data[:, -1]
-
-        self.assertTrue(phi_s_p.shape, (yz_pts**2, 1))
+    pass
```
Comment (tinosulzer) [inline tests/integration/test_models/test_full_battery_models/test_lithium_ion/test_external/test_external_current_collector.py:11]: delete this test

File: tests/integration/test_models/test_full_battery_models/test_lithium_ion/test_external/test_external_temperature.py [hunk 2]
```diff
@@ -7,86 +7,98 @@
...
-        for i in np.arange(1, len(t_eval) - 1):
-            dt = t_eval[i + 1] - t_eval[i]
-            T = (np.sin(2 * np.pi * x) * np.sin(2 * np.pi * 100 * t_eval[i]))[
-                :, np.newaxis
-            ]
-            external_variables = {"Cell temperature": T}
-            sim.step(dt, external_variables=external_variables)
+            sim.step(dt, inputs=inputs)  # works
+
+    # def test_external_temperature(self):
```
Comment (tinosulzer) [inline tests/integration/test_models/test_full_battery_models/test_lithium_ion/test_external/test_external_temperature.py:30]: delete these

File: tests/unit/test_models/test_full_battery_models/test_lithium_ion/test_spme.py [hunk 3]
```diff
@@ -10,22 +10,23 @@ class TestSPMe(BaseUnitTestLithiumIon, unittest.TestCase):
...
-        )
-
-        # a variable
-        model_options = {"thermal": "lumped", "external submodels": ["thermal"]}
-        model = pybamm.lithium_ion.SPMe(model_options)
-        self.assertEqual(
-            model.external_variables[0],
-            model.variables["Volume-averaged cell temperature"],
-        )
+    # def test_external_variables(self):
```
Comment (tinosulzer) [inline tests/unit/test_models/test_full_battery_models/test_lithium_ion/test_spme.py:13]: delete

**Review by tinosulzer — CHANGES_REQUESTED** [review]
Existing changes look good but there are more lines to remove, see coverage

**Comment by TomTranter** [comment]
@tinosulzer tests passing but some benchmarks are failing. How can I run just the failing ones locally easily?

**Review by tinosulzer — APPROVED** [review]
Thanks @TomTranter

**Pull request merged**